drivers/hmis_external_apis/app/jobs/hmis_external_apis/ac_hmis/upload_clients_job.rb

```diff
@@ -8,36 +8,123 @@
+        else
+          raise "invalid item to upload: #{mode}"
+        end
         self.state = :success
       else
         self.state = :not_run
-        Rails.logger.info 'Not running client upload due to lack of credentials'
+        Rails.logger.info "Not running #{mode} due to lack of credentials"
       end
     rescue StandardError => e
       puts e.message
       self.state = :failed
       @notifier.ping('Failure in upload clients job', { exception: e })
       Rails.logger.fatal e.message
     end
+
+    private
+
+    def clients_with_mci_ids_and_address
+      export = Exporters::ClientExport.new
+      export.run!
+
+      uploader = Exporters::ClientExportUploader.new(
+        filename_format: '%Y-%m-%d-clients.zip',
+        io_streams: [
+          OpenStruct.new(
+            name: 'Client.csv',
+            io: export.output,
+          ),
+        ],
+      )
+
+      uploader.run!
+    end
+
+    def hmis_csv_export
+      # FIXME: Not sure what this should be a hash of. I'll refactor to take content
+      # into account in a later step if that's needed. Could be a hash of the
+      # current time or content or sorted list of unique ids in the file. Or,
+      # maybe something else.
+      hash = SecureRandom.hex(8)
+      user = User.system_user
+      version = '2024'
+
+      data_source = HmisExternalApis::AcHmis.data_source
+
+      filter = ::Filters::HmisExport.new(data_source_ids: [data_source.id], version: version, user_id: user.id)
+      exporter = filter.schedule_job(report_url: warehouse_reports_hmis_exports_url, perform_now: true)
+
+      # FIXME: need to know how to get job and/or resulting export so I can poll for the results
+      sleep 1
+
+      # FIXME: I don't know how to find the export generated by the filter
+      # above or if I used the code correctly.
+      export = exporter.export
+
+      # FIXME: The story said the report will be at the url above (which I
+      # think is warehouse_reports/hmis_exports) but it seems convoluted to
+      # have to make a web request here. I think I should probably be able to
+      # use the previous line's logic somehow
+
+      # FIXME: All the files in
+      # https://files.hudexchange.info/resources/documents/HMIS-CSV-Format-Specifications-2024.pdf
+      # requested. I can't make a local export yet, so I can't confirm yet.
+
+      uploader = Exporters::ClientExportUploader.new(
+        filename_format: "%Y-%m-%d-HMIS-#{hash}-hudcsv.zip",
+        # FIXME: I believe this content is a string of zipped content, but I haven't confirmed that.
+        pre_zipped_data: export.file.read,
+      )
+
+      uploader.run!
+    end
+
+    def project_crosswalk
+      # Use these for reference:
+      # app/views/warehouse_reports/hmis_cross_walks/index.xlsx.axlsx
+      # app/controllers/warehouse_reports/hmis_cross_walks_controller.rb
+
+      # FIXME: what are we crosswalking exactly? Here's some code that might be useful:
```

Comment:
The Excel file just provides lookups (cross-walks) between warehouse identifiers (which end up in ProjectID in our exports) and the ProjectID that comes to us from the sending HMIS.